presto-docs/src/main/sphinx/functions/array.rst

```diff
@@ -129,6 +129,16 @@ Array Functions
 
     Flattens an ``array(array(T))`` to an ``array(T)`` by concatenating the contained arrays.
 
+.. function:: ngrams(array<T>, n) -> array<array<T>>
```

Comment:
Just want to double check that everyone is happy with `array` then `n` and not the other way around (we only get one shot at this)

Reply:
Putting `n` first probably makes more sense given the name "ngrams" and since it will likely be a constant and the array won't be.

Reply:
Given that other common array operations such as `element_at`, `contains` and `repeat` put the (likely) constant at the end, I think putting the constant at the end makes it more consistent with other array operations that will be beside it in our documentation.  Also, for what it's worth, the Hive UDF puts it at after the array.

Reply:
I agree with @tdcmeehan. For example element_at(array<E>, index)

Reply:
@dain What is your opinion on this?

Reply:
From a language purity perspective, it might be cleaner to use `ngrams(n, array(t))`. In a language that supports currying and higher level functions, the variant `ngrams(n)` would be a function that returns another function that computes an ngram of order `n` for any given input.

However, the `ngrams(array(t), n)` form is more consistent other functions in the library which place the object of the computation in the first argument. E.g., `word_stem(word, lang)`, `substr(string, start)`, `strpos(string, substring)`, `split(string, delimiter)`, `array_sort(array(T), function(T, T, int))`, `repeat(element, count)`, etc.

So, let's go with the current form.

Reply:
BTW, the canonical form for parametric types is with parenthesis, not angle brackets. I.e., `array(t)` instead of `array<t>`

Reply:
Thanks @martint!  The angle brackets are widely used in this doc so I keep the format